tar2rpm: add filename '-' as an alias for stdin and stdout

totph opened the pull request:
When no filenames are given the informative message
"tar2pm: reading tar content from stdin, writing rpm data to stdout."
is printed to stderr. This can be suppressed by explicitly
specifying '-' for TARFILE and RPMFILE (if only TAR- or RPMFILE is
given the message reflects that).

Also updated usage() to reflect that, and rename FILE to
TARFILE/RPMFILE.

jarondl commented on cmd/tar2rpm/main.go:
```diff
@@ -102,11 +102,15 @@ func main() {
-		f, err := os.Open(flag.Arg(0))
-		if err != nil {
-			log.Fatalf("Failed to open file %s for reading\n", flag.Arg(0))
+		if flag.Arg(0) == "-" {
```
Years from now, someone is going to be surprised by the difference between this and "case 0", and they will try to fix this.

Can you add a comment, explaining the rationale?

totph replied:
Done

jarondl replied:
I don't see the change?

jarondl reviewed (commented):
Thanks for the contribution!

I'm not sure I understand the rationale (or indeed the solution). What do you care if some message is printed to stderr? In common practice apps ignore stderr. Can you explain the reasoning?

If this is about some kind of "quietness" or "verbosity", perhaps adding a flag for that would be better than magical behavior difference when you specify "-" or not.

(I'm not saying no to this PR, I just want to understand it better).

jarondl commented:
/gcbrun

totph commented:
This tries to follow the "silence is golden" rule of most (non-Windows) commandline tools, and `-` is the common way to specify stdin on the commandline, e.g. `cat file1 - file3`. Also, `tar cv /path | tar2rpm ..` prints the list of files, but also the one line from tar2rpm. Otherwise I would only expect stderr output on error, unless explicitly requested.

When no `-` is specified then printing a notice is helpful, as in "hey, not doing anything because I'm implicitly reading data from stdin (which might be the terminal)". Unlike, say, `grep foo` which will just sit there.

So, in my opinion no `-quiet` flag is needed, but I would add a notice to `usage()`. Or let me know if you would prefer a `-quiet` or `-verbose` flag.

jarondl commented:
> When no `-` is specified then printing a notice is helpful, as in "hey, not doing anything because I'm implicitly reading data from stdin (which might be the terminal)". Unlike, say, `grep foo` which will just sit there.
> 
> So, in my opinion no `-quiet` flag is needed, but I would add a notice to `usage()`. Or let me know if you would prefer a `-quiet` or `-verbose` flag.

Sounds good. Please add a notice to `usage()` saying that passing `-` explicitly will silence the message.

totph commented:
Updated usage, and while writing the text I split the overloaded `FILE` into `TARFILE` and `RPMFILE`, and added the `-` logic to the output logic as well so it is symmetric. Pardon the slight feature creep :)
